src/people.rs

```diff
@@ -257,26 +421,56 @@ impl ContextPeopleExt for Context {
         property: T,
         value: T::Value,
     ) {
+        assert!(!T::is_derived(), "Cannot set a derived property");
         let data_container = self.get_data_container(PeoplePlugin)
             .expect("PeoplePlugin is not initialized; make sure you add a person before accessing properties");
 
-        let current_value = *data_container.get_person_property_ref(person_id, property);
-        let previous_value = match current_value {
-            Some(current_value) => current_value,
+        let current_cached_value = *data_container.get_person_property_ref(person_id, property);
+        let previous_value = match current_cached_value {
+            Some(value) => value,
             None => {
-                let initialize_value = T::initialize(self, person_id);
+                let initialize_value = T::compute(self, person_id);
                 data_container.set_person_property(person_id, property, initialize_value);
                 initialize_value
             }
         };
 
+        // Temporarily remove dependency properties since we need mutable references
+        // to self during callback execution
+        let deps_temp = {
+            let data_container = self.get_data_container(PeoplePlugin).unwrap();
+            let mut dependencies = data_container.dependency_map.borrow_mut();
+            dependencies.get_mut(&TypeId::of::<T>()).map(std::mem::take)
+        };
+
+        let mut dependency_event_callbacks = Vec::new();
+        if let Some(mut deps) = deps_temp {
+            // If there are dependencies, set up a bunch of callbacks with the
+            // current value
+            for dep in &mut deps {
+                dep.add_event_callback(self, person_id, &mut dependency_event_callbacks);
+            }
+
+            // Put the dependency list back in
+            let data_container = self.get_data_container(PeoplePlugin).unwrap();
+            let mut dependencies = data_container.dependency_map.borrow_mut();
+            dependencies.insert(TypeId::of::<T>(), deps);
+        }
+
+        // Update the main property and send a change event
+        let data_container = self.get_data_container(PeoplePlugin).unwrap();
+        data_container.set_person_property(person_id, property, value);
         let change_event: PersonPropertyChangeEvent<T> = PersonPropertyChangeEvent {
             person_id,
             current: value,
             previous: previous_value,
         };
-        data_container.set_person_property(person_id, property, value);
         self.emit_event(change_event);
+
+        // If there are dependency callbacks, call them with the updated value
```

Comment:
You're not calling them with the updated value, though, you're just calling them and they get the updated value on their own.